src/image/ImageSecondary.java

```diff
@@ -237,47 +238,54 @@ public boolean equals(Object o) {
-        blurredImage.copyFrom(this); // Black Bar Issue
+        int height = this.getHeight();
+        int width = this.getWidth();
+        int[][][] original = this.getPixels();
 
-        for (int r = 1; r < this.getHeight(); ++r) {
-            for (int c = 1; c < this.getWidth(); ++r) {
-                // Average Each Color Value
-                int redSum = this.getPixels()[r][c][0]
-                + this.gettotalPixel()[r - 1][j - 1][0];
-                int greenSum = this.getPixels()[r][c][1]
-                + this.gettotalPixel()[r - 1][j - 1][1];
-                int blueSum = this.getPixels()[r][c][2]
-                + this.gettotalPixel()[r - 1][j - 1][2];
+        // Define 3x3 Gaussian kernel, normalized
+        double[][] kernel = { { 1 / 16.0, 2 / 16.0, 1 / 16.0 },
+                { 2 / 16.0, 4 / 16.0, 2 / 16.0 },
+                { 1 / 16.0, 2 / 16.0, 1 / 16.0 } };
 
-                int redAvg = redSum / GUSSAINDIVI;
-                int greenAvg = greenSum / GUSSAINDIVI;
-                int blueAvg = blueSum / GUSSAINDIVI;
+        // Create a new Image to store blurred pixels
+        Image blurred = new Image1(width, height, new int[] { 0, 0, 0 });
 
-                // Set Current Pixel to the Result of the Blur
-                blurredImage[r][c][0] = redAvg;
-                bluredImage[r][c][1] = greenAvg;
-                blurredImage[r][c][2] = blueAvg;
-            }
-        }
+        for (int i = 0; i < height; ++i) {
+            for (int j = 0; j < width; ++j) {
+                int[] blurredPixel = new int[3];
 
-        // Repalce Values
-        for (int r = 0; r < this.getHeight(); ++r) {
-            for (int c = 0; c < this.getWidth(); ++r) {
-                // Average Each Color Value
-                this.getPixels()[r][c][0] = blurredImage[r][c][0];
-                this.getPixels()[r][c][1] = blurredImage[r][c][1];
-                this.getPixels()[r][c][2] = blurredImage[r][c][2];
+                for (int channel = 0; channel < 3; ++channel) {
+                    double sum = 0.0;
+
+                    for (int ki = -1; ki <= 1; ++ki) {
+                        for (int kj = -1; kj <= 1; ++kj) {
+                            int ni = i + ki;
+                            int nj = j + kj;
+
+                            if (ni >= 0 && ni < height && nj >= 0
+                                    && nj < width) {
+                                sum += original[ni][nj][channel]
+                                        * kernel[ki + 1][kj + 1];
+                            }
+                        }
+                    }
+
+                    blurredPixel[channel] = this.clip((int) Math.round(sum));
+                }
+
+                int index = i * width + j;
+                blurred.paint(index, blurredPixel);
             }
         }
+
+        // Replace current image with blurred one
+        this.copyFrom(blurred);
     }
 }
```

Comment:
I think I forgot to mention, but make sure you implement `equals()`, `toString()`, and `hashCode()` here. You can exclude hashCode if it doesn't make sense, but I think hashing an image would be useful.

Reply:
Nevermind! I see them buried in the code.